include/core/rtos/Queue.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef EVT_RTOS_QUEUE_
+#define EVT_RTOS_QUEUE_
+
+#include <core/rtos/Initializable.hpp>
+#include <functional>
+
+namespace core::rtos {
+
+/**
+ * Class that wraps a ThreadX Queue. Queues are a FIFO data structure, and threadx queues provide additional
+ * functionality for interacting with queues in a threaded environment.\n\n
+ *
+ * This class extends Initializable, and like all other Initializable classes must be passed into the
+ * Threadx::startKernel() method as part of the initList parameter.
+ */
+class Queue : public Initializable {
+public:
+    /**
+     * Constructs a Queue object, but does not initiaize it (must call init before using).\n\n
+     *
+     * NOTE: It is technically possible to overflow the size of the queue. If (messageSize * numMessages) is greater
+     * than the max value of a uint32_t (roughly 4.2 billion), the calculation will overflow and your queue will be
+     * incorrectly sized. Take care to ensure this does not happen.
+     *
+     * @param[in] name The name of the queue.
+     * @param[in] messageSize Size (in 4-byte words) of each message in the queue. The message size can be at most 16.
+     * @param[in] numMessages Number of messages in the queue.
```

Comment:
^

Reply:
Updated the comment so that it's slightly more clear, but Matt it's the number of messages the queue can fit. The variable exactly describes what it means.